Add crayons_icon_tag helper

citizen428 opened the pull request:
## What type of PR is this? (check all applicable)

- [X] Feature

## Description

This PR adds a new `crayons_icon_tag` helper that wraps the existing `inline_svg_tag` helper with icon-specific options. 

## Related Tickets & Documents

See https://forem.team/pp/icons-in-crayons-30k for details.

## QA Instructions, Screenshots, Recordings

Nothing specific, specs should be green

![Screen Shot 2564-12-28 at 10 42 01](https://user-images.githubusercontent.com/47985/147525706-6e84cbb8-5bf6-422d-9008-1dbe3338bbf3.png)

### UI accessibility concerns?

None

## Added/updated tests?

- [X] Yes

## [Forem core team only] How will this change be communicated?

- [X] I will share this change internally with the appropriate teams

citizen428 commented on app/helpers/application_helper.rb:
```diff
@@ -322,4 +322,19 @@ def creator_settings_form?
+    icon_class = "crayons-icon"
+    icon_class = "#{icon_class} #{css_class}" if css_class
+
+    inline_svg_tag(icon_name, aria: true, width: 24, height: 24, class: icon_class, **opts)
```
I've included `aria: true` by default since this adds `role="img"`. It also seems like we pretty much manually added this for all our existing icons anyway.

ludwiczakpawel commented:
You're the hero we don't deserve! <3 

Thanks so much buddy! Two quick questions:

1. Is there a chance to add `native` parameter so it would work similar to the JSX component? 
2. This is more me just thinking out loud: would it make sense to have some shorter more generic name, like just `icon` or `crayons_icon`. Again, not a big deal, just wondering...

citizen428 commented:
> Is there a chance to add native parameter so it would work similar to the JSX component?

I'll look at the JSX component because I frankly don't know what this means 😅 

> would it make sense to have some shorter more generic name

I chose the name to be consistent with other Rails helper methods, i.e. using a `_tag` suffix for methods that directly return markup (like the wrapped `inline_svg_tag` helper). The reason I prefixed it with `crayons_` is that I want to make sure people understand that this is a method under our control that we can change as needed, which is harder to infer from a generic name like `icon_tag`.

citizen428 commented:
@ludwiczakpawel Added the `native` option, please sanity check. Also updated the method to now accept symbol names, e.g. `crayons_icon_tag(:twitter)` which feels more idiomatic in Ruby. I also moved the code into a new `CrayonsHelper` but that's more of a code organization thing.

djuber reviewed (approved):
Approving, but I assume you're waiting for Pawel's feedback/review

citizen428 commented:
@djuber I am. In theory, we could merge and do a follow-up since I haven't changed any code to make use of the new helper yet. But since this isn't exactly urgent I think it's ok to wait.

ludwiczakpawel reviewed (approved):
<3<3<3
